server/src/com/thoughtworks/go/server/service/ArtifactsService.java

```diff
@@ -390,7 +401,7 @@ private boolean deleteFile(File file) {
     }
 
     public void appendToConsoleLog(JobIdentifier jobIdentifier, String text) throws IllegalArtifactLocationException, IOException {
-        File file = findArtifact(jobIdentifier, ArtifactLogUtil.getConsoleLogOutputFolderAndFileName());
```

Comment:
Also, there is almost implicit knowledge hidden in some places, right? Like the fact that ```chooser.findArtifact``` can be costly if the artifact store is on the network. So, try to find the temporary console log location first, rather than trying both, etc.